hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -49,78 +53,109 @@
+	private final TransactionalLedger<Pair<AccountID, TokenID>, TokenRelProperty, MerkleTokenRelStatus> tokenRelsLedger;
+	private final SideEffectsTracker sideEffectsTracker;
+	private final TokenStore tokenStore;
+	private final MerkleAccountScopedCheck scopedCheck;
+	private final UniqTokenViewsManager tokenViewsManager;
+	private final AutoAccountCreateLogic autoAccountCreator;
 
-    @Inject
-    public TransferLogic(TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accountsLedger,
-                         TransactionalLedger<NftId, NftProperty, MerkleUniqueToken> nftsLedger,
-                         TransactionalLedger<Pair<AccountID, TokenID>, TokenRelProperty, MerkleTokenRelStatus> tokenRelsLedger,
-                         TokenStore tokenStore,
-                         SideEffectsTracker sideEffectsTracker,
-                         UniqTokenViewsManager tokenViewsManager,
-                         GlobalDynamicProperties dynamicProperties,
-                         OptionValidator validator) {
-        this.accountsLedger = accountsLedger;
-        this.nftsLedger = nftsLedger;
-        this.tokenRelsLedger = tokenRelsLedger;
-        this.sideEffectsTracker = sideEffectsTracker;
-        this.tokenStore = tokenStore;
-        this.tokenViewsManager = tokenViewsManager;
+	@Inject
+	public TransferLogic(final TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accountsLedger,
+			final TransactionalLedger<NftId, NftProperty, MerkleUniqueToken> nftsLedger,
+			final TransactionalLedger<Pair<AccountID, TokenID>, TokenRelProperty, MerkleTokenRelStatus> tokenRelsLedger,
+			final TokenStore tokenStore,
+			final SideEffectsTracker sideEffectsTracker,
+			final UniqTokenViewsManager tokenViewsManager,
+			final GlobalDynamicProperties dynamicProperties,
+			final OptionValidator validator,
+			final AutoAccountCreateLogic autoAccountCreator) {
+		this.accountsLedger = accountsLedger;
+		this.nftsLedger = nftsLedger;
+		this.tokenRelsLedger = tokenRelsLedger;
+		this.sideEffectsTracker = sideEffectsTracker;
+		this.tokenStore = tokenStore;
+		this.tokenViewsManager = tokenViewsManager;
+		this.autoAccountCreator = autoAccountCreator;
 
-        scopedCheck = new MerkleAccountScopedCheck(dynamicProperties, validator);
-    }
+		scopedCheck = new MerkleAccountScopedCheck(dynamicProperties, validator);
+	}
 
-    public void transfer(List<BalanceChange> changes) {
-        var validity = OK;
-        for (var change : changes) {
-            if (change.isForHbar()) {
-                validity = accountsLedger.validate(change.accountId(), scopedCheck.setBalanceChange(change));
-            } else {
-                validity = tokenStore.tryTokenChange(change);
-            }
-            if (validity != OK) {
-                break;
-            }
-        }
+	public void transfer(final List<BalanceChange> changes) {
+		var validity = OK;
+		List<ByteString> autoCreationAliases = new ArrayList<>();
```

Comment:
We should wait to create this `ArrayList` until we know an auto-creation is being attempted; creating an `ArrayList` is actually fairly heavy, so we don't want to do it unnecessarily 10,000 times a second. 😁

Reply:
Ah yes . Modified to create new list only if accounts to be auto created 😛